### PR TITLE
Refactor error handle

### PR DESCRIPTION
**Pull request by ttaoS**

### CODE REVIEW

File: src/ts/services/keyring-vault-proxy.ts [hunk 1]
```diff
@@ -43,9 +43,13 @@ export function unlockWallet (password: string, addressPrefix?: string):
   Promise<Array<KeyringPair$Json>> {
   return new Promise<Array<KeyringPair$Json>>(resolve => {
     port.onMessage.addListener(msg => {
-      if (msg.method === FUNCS.UNLOCK) {
-        resolve(msg.result)
+      if (msg.method !== FUNCS.UNLOCK) return
+      debugger
```
Comment (fyang1024) [inline src/ts/services/keyring-vault-proxy.ts]: should we remove the debugger?

File: src/ts/components/account/ConfirmPhrase.tsx [hunk 2]
```diff
@@ -68,9 +67,9 @@ class ConfirmPhrase extends React.Component<IConfirmPhraseProps, IConfirmPhraseS
             console.log('Account created! ', keyringPair)
             this.props.setCreated(true)
             this.setState({ keyringPair: keyringPair })
-          })
+          }).catch(err => { this.props.setError(err) })
```
Comment (mytechtip) [inline src/ts/components/account/ConfirmPhrase.tsx]: In the catch statement, is `err` an object or a string?

Comment (ttaoS) [reply src/ts/components/account/ConfirmPhrase.tsx]: it is a message string.

File: src/ts/services/keyring-vault-proxy.ts [hunk 3]
```diff
@@ -18,45 +18,53 @@ export function isWalletLocked (): Promise<boolean> {
...
 export function unlockWallet (password: string, addressPrefix?: string):
   Promise<Array<KeyringPair$Json>> {
-  return new Promise<Array<KeyringPair$Json>>(resolve => {
+  return new Promise<Array<KeyringPair$Json>>((resolve, reject) => {
     port.onMessage.addListener(msg => {
-      if (msg.method === FUNCS.UNLOCK) {
-        resolve(msg.result)
+      if (msg.method !== FUNCS.UNLOCK) return
+      if (msg.error) {
+        reject(msg.error.message)
```
Comment (ttaoS) [inline src/ts/services/keyring-vault-proxy.ts]: this is the place we return message rather object @mytechtip

**Pull request merged**